### ISSUE TITLE
Install packages from Gitlab and Gitea instances

### ISSUE THREAD

**Issue by trymeouteh** [npm/cli#5973]
Gitlab and Gitea allows you to self host your own git servers. It would be amazing to see the ability to install packages directly from any gitlab or gitea instance.

And to have the ability to have packages use dependancies directly from Github, Gitlab and Gitea repository and not have to only rely on dependencies from the NPM repository.

Such a feature will allow for decentralisation, private repositories and the ability to truly self host your own packages.

**Comment by ljharb** [comment]
You can already, and have always been able to - you can npm install a URL from any git repository located anywhere.

That said, it's considered a bad practice to have any non-registry dependencies.

**Comment by trymeouteh** [comment]
> You can already, and have always been able to - you can npm install a URL from any git repository located anywhere.

I cannot find a working example of a package on a gitea instance to install with NPM from the gitea repository.

> That said, it's considered a bad practice to have any non-registry dependencies.

And you are saying it is possible to make packages have dependencies on git repositories?

**Comment by ljharb** [comment]
Yes, it's always been possible - it's just a bad idea. Git is for development, not consumption.

I'm not familiar with Gitea but the first page of google results links me to https://discourse.gitea.io/t/use-gitea-to-host-npm-modules/933, which seems like an authoritative source.
